code/modules/modular_computers/file_system/programs/command/comm.dm

```diff
@@ -289,15 +267,17 @@ var/last_message_id = 0
 	for (var/datum/comm_message_listener/l in comm_message_listeners)
 		l.Add(message)
 
-	//Old console support
-	for (var/obj/machinery/computer/communications/comm in SSmachinery.processing_machines)
-		if (!(comm.stat & (BROKEN | NOPOWER)) && comm.prints_intercept)
-			var/obj/item/weapon/paper/intercept = new /obj/item/weapon/paper( comm.loc )
-			intercept.name = message_title
-			intercept.info = message_text
-
-			comm.messagetitle.Add(message_title)
-			comm.messagetext.Add(message_text)
+	for (var/l in GET_LISTENERS("modular_computers"))
+		var/listener/L = l
+		if(istype(L.target, /obj/item/modular_computer/console/preset/command/main))
```

Comment:
This should not iterate through a specific preset - add a listener on the command&control program itself so that programs that download it get it too.